Sprint 11

pitinovamargo opened the pull request.

pitinovamargo committed:
Заблокирована возможная гонка при авторизации

pitinovamargo committed:
Перенесено получение данных профиля в сплеш-скрин, сделано обновление лейблов при авторизации

pitinovamargo committed:
Добавлено получение аватарки пользователя

пока не доработано до самого получения, этап поиска ошибки

pitinovamargo committed:
Получен URL аватара пользователя

pitinovamargo committed:
Добавлены нотификации при получении URL аватара профиля

pitinovamargo committed:
Заменено «старое» API для KVO на «новое» в WebViewViewController

pitinovamargo committed:
Заменено использование dataTask на objectTask с использованием дженерик-ответа

pitinovamargo committed:
Сделан показ алерта при сетевых ошибках. Пока что сразу за алертом идет переключение на tabBarController

pitinovamargo committed:
Добавлено появление алерта в случае сетевых ошибок

pitinovamargo committed:
Применен Kingfisher для загрузки аватарки профиля

pitinovamargo committed:
Реализовано сохранение токена в Keychain вместо UserDefaults

pitinovamargo committed:
Сверстан в коде ProfileViewController

pitinovamargo committed:
Сверстан в коде SplashViewController

Но пока есть ошибка -> не переходит на AuthViewController (черный экран)

pitinovamargo committed:
Исправлено создание AuthViewController в коде, пофикшено отображение аватара в форме круга без фоновых углов

Dmitrii-Diadiushkin commented on ImageFeed/Auth/SplashViewController.swift:
```diff
@@ -47,20 +59,55 @@ class SplashViewController: UIViewController {
 
 extension SplashViewController: AuthViewControllerDelegate {
     func acceptToken(code: String) {
```
названия методов лучше всего начинать с глагола, который описывает действие метода. Так при чтении кода легче понять, что метод делает.

Dmitrii-Diadiushkin commented on ImageFeed/Auth/SplashViewController.swift:
```diff
@@ -47,20 +59,55 @@ class SplashViewController: UIViewController {
+                self.profileService.fetchProfile(accessToken)
+                self.profileImageService.fetchProfileImageURL(username: self.profileService.getProfile()?.username ?? "") { result in
+                    switch result {
+                    case .failure(_):
```
в этом случае не происходит разблокировки экрана. И алерт не показывает (из-за того, что поверх экрана Splash еще есть экраны auth и webView - их надо убрать, чтобы алерт нормально отобразился)

Dmitrii-Diadiushkin commented on ImageFeed/Profile/ProfileService.swift:
```diff
@@ -0,0 +1,85 @@
+struct ProfileResult: Codable {
+    let username: String
+    let firstName: String
+    let lastName: String
```
lastName может отсутствовать

pitinovamargo committed:
Исправлена ситуация, когда нет фамилии пользователя

Также реализована попытка показать алерт при проблеме с авторизацией (пока попытка неудачная)

pitinovamargo committed:
Удален KeychainWrapper, используемый для тестирования

pitinovamargo committed:
TMP

pitinovamargo committed:
Revert "TMP"

This reverts commit 0ffe005adcd0847e7666f737d5f24bf5ba89ce79.

pitinovamargo committed:
Исправлен показ алерта в случае ошибки

Dmitrii-Diadiushkin commented on ImageFeed/Auth/SplashViewController.swift:
```diff
@@ -47,20 +59,54 @@ class SplashViewController: UIViewController {
+                    self.profileImageService.fetchProfileImageURL(username: self.profileService.getProfile()?.username ?? "") { result in
+                        switch result {
+                        case .failure(_):
+                            UIBlockingProgressHUD.dismiss()
```
UIBlockingProgressHUD.dismiss() - эта строка вызывается в обоих случаях. Лучше вынести ее за пределы switch case и вызвать один раз, чтоб избежать дублирования кода
